### ISSUE TITLE
asyncftpclient - Error: 'defaultOnProgressChanged' cannot be passed to a procvar

### ISSUE THREAD

**Issue by pyloor** [nim-lang/Nim#4681]
Hi,
the asyncftpclient example does not compile:

```nim
import asyncdispatch, asyncftpclient

proc main() {.async.} =
  var ftp = newAsyncFtpClient("example.com", user = "test", pass = "test")
  await ftp.connect()
  let currentDir = await ftp.pwd()
  assert currentDir == "/home/user/"
  await ftp.store("file.txt", "file.txt")
  echo("File finished uploading")

waitFor(main())
```

Crashes with
```nim
lib/pure/asyncftpclient.nim(342, 54) Error: 'defaultOnProgressChanged' cannot be passed to a procvar
```

Nim version:
```nim
Nim Compiler Version 0.14.3 (2016-06-18) [Linux: amd64]
Copyright (c) 2006-2016 by Andreas Rumpf

git hash: 8182414a6f99c46dc31ba63e00fe1315981d2b50
active boot switches: -d:release
```

**Comment by dom96** [comment]
I think fixing this is fairly simple, you have to add the ``procvar`` pragma somewhere.

**Comment by pyloor** [comment]
Thanks for your hint. I have added in asyncftpclient.nim the procvar to the defaultOnProgressChanged and the compilation does not crash.
```nim
proc defaultOnProgressChanged*(total, progress: BiggestInt,
    speed: float): Future[void] {.nimcall,gcsafe,procvar.} =
  ## Default FTP ``onProgressChanged`` handler. Does nothing.
  result = newFuture[void]()
  #echo(total, " ", progress, " ", speed)
  result.complete()
```
But I am currently not sure this is right :-/

**Comment by dom96** [comment]
Assuming this was fixed by #4780.
